src/ui/public/vis/update_status.ts

```diff
@@ -29,6 +29,31 @@ enum Status {
   UI_STATE = 'uiState',
 }
 
+/**
+ * Checks whether the hash of a specific key in the given oldStatus has changed
+ * compared to the new valueHash passed.
+ */
+function hasHashChanged<T extends string>(
+  oldStatus: { [key in T]?: string },
+  name: T,
+  valueHash: string
+): boolean {
+  const oldHash = oldStatus[name];
+  return oldHash !== valueHash;
+}
+
+interface Size {
+  width: number;
+  height: number;
+}
+
+function hasSizeChanged(size: Size, oldSize?: Size): boolean {
```

Comment:
Could you invert the params here so we are a bit more aligned with the previous: `hasHashChanged` function?
first the oldStatus and the new one?

Reply:
Unfortunately I can't because `oldSize` is optional so it must be in the end. If you want I can change the parameter order to the `hasHashChanged` function if you want to `hash, oldStatus, key` if you think that increases readability.